ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -1136,19 +1188,22 @@ def get_speedup_time(self, start, end):
         jerk = jmt.get_j_at(0.1)
         acc = jmt.get_a_at(T - 0.5)
 
-        rospy.logdebug("found jerk of {:3.2f} with a_dist of {:3.2f}"
-                       " and acc of {:3.2f} with speedup time of {:3.2f}"
-                       .format(jerk, a_dist, acc ,T))
+        rospy.logdebug("found initial jerk={:3.2f}m/s^3 and final acc = {:3.2f}m/s^2 "
+                       "when using a_dist={:3.2f}m and T={:3.2f}s"
+                       .format(jerk, acc, a_dist, T))
                 
         optimized = False
         time_diff = T * 0.01
         counter = 0
         if acc < 0.0:
             while optimized is False:
-                T = T - time_diff
+                T = T + time_diff
```

Comment:
This had been searching in the wrong direction

Reply:
Shouldn't it be looking for a shorter time though?  In the case of slowing down from an initial positive accel, the initial overspeed causes the car to cover extra distance so the JMT has to compensate by adding extra slowdown that overshoots the end speed target.  In the case of speeding up from an initial negative accel, the initial underspeed should cause the car to cover less distance so the JMT would compensate by adding extra speed that overshoots the end speed targe, so shortening the time cuts off this extra speed for the same distance.  Maybe my intuition is not correct if you saw that it actually needs longer time in testing.

Reply:
After testing and observing, here are my conclusions:
if you follow argument to the limit of t->0, there will be almost no time to make the change in speed so the car will have to accelerate very quickly then slow down again quickly to be able to cover the distance in the allotted time.  Increasing the time may cause the curve to stay slower longer before increasing the speed at the end. As long as there is a reasonable estimate of time and the distance is reasonably long, we have not had the issue with the time being too long.  

I will think about  a check to see if this is the case but for now this has worked in all testing so I will not put in a change on the merge.

idea: compare magnitude of calculated accel between loops and see if improving or not and possibly change direction of search.

Reply:
I see, it can be tricky on either side.  No problem to leave it since it's working.